mutex.go

```diff
@@ -44,10 +44,21 @@ var (
 // Lock waits up to duration d for l.Lock() to succeed.
 func Lock(l sync.Locker, d time.Duration) error {
 	done := make(chan struct{}, 1)
-	go func() {
-		l.Lock()
-		done <- struct{}{}
-	}()
+	m, ok := l.(*mutex)
+	// if l is a mutex, lock with a timeout context.
+	if ok {
+		ctx, cancel := context.WithTimeout(context.Background(), d*time.Second)
```

Comment:
next commit?... hm, I think github is showing me the last commit, unless I don't know how to use this new review tool.